runtime/lua/vim/_defaults.lua

```diff
@@ -143,6 +145,35 @@ do
     end
     vim.keymap.set({ 'o' }, 'gc', textobject_rhs, { desc = 'Comment textobject' })
   end
+
+  --- Map [d and ]d to move to the previous/next diagnostic. Map <C-W>d to open a floating window
+  --- for the diagnostic under the cursor.
+  ---
+  --- See |[d-default|, |]d-default|, and |CTRL-W_d-default|.
+  do
+    vim.keymap.set('n', ']d', function()
+      vim.diagnostic.goto_next({ float = false })
+    end, {
+      desc = 'Jump to the next diagnostic with the highest severity',
+    })
+
+    vim.keymap.set('n', '[d', function()
+      vim.diagnostic.goto_prev({ float = false })
+    end, {
+      desc = 'Jump to the previous diagnostic with the highest severity',
+    })
+
+    vim.keymap.set('n', '<C-W>d', function()
+      vim.diagnostic.open_float({ border = 'rounded' })
+    end, {
+      desc = 'Open a floating window showing diagnostics under the cursor',
+    })
+
+    vim.keymap.set('n', '<C-W><C-D>', '<C-W>d', {
+      remap = true,
```

Comment:
nit: would be safer to reference the same local function instead of remapping. Not a blocker.